### PR TITLE
Add test cases to verify ps-AF(Pashto-Afghanistan), ps-PK(Pashto-Pakistan) locales 

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Added test cases to verify ps-AF(Pashto-Afghanistan), ps-PK(Pashto-Pakistan) 
   *  Default calendar: ps-AF(persian), ps-PK(gregorian)
   *  cldr tool code fixing : gendateformats.2.js, datefmt.js, genlislt.js. 
   *  dateformats.json and sysres.json files are re-generated which were missing persin data info and also there was a bug.
* Added test case feature: LocaleInfo,Script, String(plural) LocaleMatch, DateFmt, DurFmt, DateRngFmt, List, Unit, firstDayofWeek, clock, calendar,Month/Weekday Translation, NumFmt,
* or translation and some unitfmt translations are manually updated by google translate's help for ListFmt

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

**Comment by gouniLee** [comment]
@ehoogerbeets waiting for your review.

**Comment by gouniLee** [comment]
@ehoogerbeets  Due to a late review, the schedule that I have to meet has been delaying. I need the new version of iLib including this PR by this week.

File: js/data/locale/ps/sysres.json [hunk 1]
```diff
@@ -121,6 +173,38 @@
     "LLL12": "دسمبر",
     "LL12": "دس",
     "L12": "12",
+    "EEEE0": "يونۍ",
+    "EEE0": "يونۍ",
+    "EE0": "يونۍ",
+    "E0": "S",
```
[the remark this answers is not in the document]
Comment (gouniLee) [reply js/data/locale/ps/sysres.json]: I updated it.

File: tools/cldr/gendatefmts2.js [hunk 2]
```diff
@@ -174,6 +174,22 @@ var hardCodeData = {
         "#{num}w": "one#{num} v|#{num} v",
         "durationShortMonths": "one#{num} ng|#{num} ng",
         "#{num}y": "one#{num} nk|#{num} nk"
+    },
+    "ps": {
+        "E0": "ﺡ",//copied data from ar/sysres.json
```
Comment (ehoogerbeets) [inline tools/cldr/gendatefmts2.js:179]: Arabic and Farsi are the wrong languages. Same script, different language. You should take the first character of the Pashto translation for the abbreviation.

Comment (gouniLee) [reply tools/cldr/gendatefmts2.js:179]: As far as I know, Arabic changes its form depending on the position.  and not sure if it's a good idea to take the first character and even sure it's correct or not. If so should we just follow CLDR? I've found there are the same case already. uz/Arab/sysres.json

Comment (gouniLee) [reply tools/cldr/gendatefmts2.js]: As we discussed, I roll back my changes. Please check them again.

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**